tests/test_completion.py

```diff
@@ -53,6 +53,21 @@ def create_config_loader() -> ConfigLoaderImpl:
     )
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
+class TestDotInPathCompletion:
+    def test_bash_completion_with_dot_in_path(self) -> None:
+        process = subprocess.Popen(
+            "tests/scripts/test_bash_dot_in_path.bash",
+            shell=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        stdout, stderr = process.communicate()
+        assert stderr == b""
+        assert stdout == b"TRUE\n"
+        return
```

Comment:
if exit with an error, this entire thing becomes something like:

```python
subprocess.check_returncode("/bin/bash", "tests/scripts/test_bash_dot_in_path.bash") == 0
```

Reply:
Checking `stdout` could have done in the bash script itself - I agree - but checking for **anything** in `stderr` is something which I like to do in my tests to capture things which I haven't planned for.
I'm also not doing any interactivity here - but may be in the future it will be needed and `popen.communicate` will be there waiting for me.
Both of the points above gives me a feeling that it's better to stay as it is - but if you want - I can recode it - just let me know.

Reply:
sure, we can go with this for this case.

for Python calls I have run_with_error() that specifically return the stderr.
I have yet  to encounter a case when I wanted to test both stdout and stderr.